File: python/7 kyu/92.py
# https://www.codewars.com/kata/67757660c552a3a7ef9aaceb/train/python

# Passed

from string import ascii_uppercase

def validate_base(num: str, base: int) -> bool:
    base_string = "0123456789" + ascii_uppercase
    current_base_digits= set(base_string[:base]) # will let us know which all digits are valid for current base
    
    for digit in num:
        if digit not in current_base_digits:
            # num is not valid for current base
            return False

    return True
    
output = validate_base('6124', 5)
print(output)

File: nodejs/7 kyu/92.js
// https://www.codewars.com/kata/55b1fd84a24ad00b32000075/train/javascript

// Passed

var AmIAfraid = function (day, num) {
  const dayNumberMap = {
    Monday: (num) => num === 12,
    Tuesday: (num) => num > 95,
    Wednesday: (num) => num === 34,
    Thursday: (num) => num === 0,
    Friday: (num) => num % 2 === 0,
    Saturday: (num) => num === 56,
    Sunday: (num) => num === 666 || num === -666,
  };

  const isAfraid = dayNumberMap[day](num);
  return isAfraid;
};

const output = AmIAfraid("Tuesday", 965);
console.log(output);